### ISSUE TITLE
shiny app doesn't run properly

### ISSUE THREAD

**Issue by tbrown122387** [mcsiple/mmrefpoints#56]
I don't think the Shiny app is working correctly. It opens up something in my web browser on localhost, but it looks like this:

![Screenshot from 2021-12-27 18-19-37](https://user-images.githubusercontent.com/1740324/147513198-e4cae0b1-6db2-4ff5-b7d6-b6c8bef72cae.png)

The console running it in the background throws the following warnings:

```
> mmrefpoints::run_app()
Loading required package: shiny

Listening on http://127.0.0.1:5687
[1] "Language change: en"
Warning: Error in $: $ operator is invalid for atomic vectors
  [No stack trace available]
Warning: Error in $: $ operator is invalid for atomic vectors
  [No stack trace available]
Warning: Error in $: $ operator is invalid for atomic vectors
  [No stack trace available]

```

openjournals/joss-reviews#3888

**Comment by mcsiple** [comment]
Interesting...I've gotten this error before when I've had issues with the `shiny.i18n` translation package; I'll have to poke around. You're using a Linux machine, right?

**Comment by tbrown122387** [comment]
Yes that’s right @mcsiple

**Comment by mcsiple** [comment]
I have to download Ubuntu and try to run the app through a Linux VM to test everything out, so it might take a little bit for me to figure out the issue-- if there are other parts of the package you can test out (individual functions, maybe) in the meantime, I think doing those first will be the most efficient way to use our time! Sorry about this; I had not tested it on Linux before.

**Comment by tbrown122387** [comment]
It might be related to one of these messages received when I run `devtools::check()`. This checks if your package is CRAN ready, and I know you're not doing that, so it will likely include warnings and notes that you're not interested in. 

```
── R CMD check results ───────────────────────────── mmrefpoints 0.0.0.9000 ────
Duration: 1m 36.8s

❯ checking R files for non-ASCII characters ... WARNING
  Found the following files with non-ASCII characters:
    00_setup.R
    13_pop_vs_yield.R
    15_plot_bycatch_guesses.R
    17_plot_proj.R
    18_multiplot_proj.R
    app_server.R
    app_ui.R
  Portable packages must use only ASCII characters in their R code,
  except perhaps in comments.
  Use \uxxxx escapes for other characters.

❯ checking package dependencies ... NOTE
  Imports includes 34 non-default packages.
  Importing from so many packages makes the package vulnerable to any of
  them becoming unavailable.  Move as many as possible to Suggests and
  use conditionally.

❯ checking for hidden files and directories ... NOTE
  Found the following hidden files and directories:
    .zenodo.json
    .github
  These were most likely included in error. See section ‘Package
  structure’ in the ‘Writing R Extensions’ manual.

❯ checking top-level files ... NOTE
  Non-standard files/directories found at top level:
    ‘CITATION.cff’ ‘paper.bib’ ‘paper.md’

❯ checking package subdirectories ... NOTE
  Found the following CITATION file in a non-standard place:
    CITATION.cff
  Most likely ‘inst/CITATION’ should be used instead.

❯ checking R code for possible problems ... NOTE
  abund_rel: no visible binding for global variable ‘median’
  app_server: no visible binding for global variable ‘lh’
  app_server: no visible binding for global variable ‘Code’
  app_server: no visible binding for global variable ‘S0’
  app_server: no visible binding for global variable ‘S1plus’
  app_server: no visible binding for global variable ‘AgeMat’
  app_server: no visible binding for global variable ‘nages’
  app_server: no visible binding for global variable ‘fmax’
  app_server: no visible binding for global variable ‘lambdaMax’
  app_server: no visible binding for global variable ‘K1plus’
  app_server: no visible binding for global variable ‘Type’
  app_server: no visible binding for global variable ‘dat’
  app_server: no visible binding for global variable ‘BycatchLvl’
  app_server: no visible binding for global variable ‘abund10’
  app_server: no visible binding for global variable ‘abund50’
  app_server: no visible binding for global variable ‘Depl’
  app_server: no visible binding for global variable ‘variable’
  app_server: no visible binding for global variable ‘value’
  app_server : <anonymous>: no visible binding for global variable
    ‘prebuild50’
  app_server : <anonymous>: no visible binding for global variable
    ‘abundrel50’
  app_server : <anonymous>: no visible binding for global variable ‘name’
  app_server : <anonymous>: no visible binding for global variable
    ‘depletion’
  app_server : <anonymous>: no visible binding for global variable ‘zero’
  app_server: no visible binding for global variable ‘depletion’
  app_server: no visible binding for global variable ‘fishing.rate’
  app_server: no visible binding for global variable ‘probability’
  app_server: no visible binding for global variable ‘Attempt’
  app_server : <anonymous>: no visible binding for global variable
    ‘Bycatch’
  app_server : <anonymous>: no visible binding for global variable
    ‘prebuild100’
  app_server : <anonymous>: no visible binding for global variable
    ‘abundrel10’
  app_server : <anonymous>: no visible binding for global variable
    ‘abundrel20’
  get_diff: no visible binding for global variable ‘S0.w’
  get_diff: no visible binding for global variable ‘S1plus.w’
  get_diff: no visible binding for global variable ‘AgeMat.w’
  get_diff: no visible binding for global variable ‘nages.w’
  get_diff: no visible binding for global variable ‘A.w’
  get_diff: no visible binding for global variable ‘z.w’
  ggradar: no visible binding for global variable ‘x’
  ggradar: no visible binding for global variable ‘y’
  ggradar: no visible binding for global variable ‘text’
  ggradar: no visible binding for global variable ‘axis.no’
  jumbotron2: no visible binding for global variable ‘button_label’
  multiplot_proj: no visible binding for global variable ‘high’
  multiplot_proj: no visible binding for global variable ‘low’
  multiplot_proj: no visible global function definition for ‘arrange’
  multiplot_proj: no visible global function definition for ‘desc’
  multiplot_proj: no visible binding for global variable ‘name’
  multiplot_proj: no visible binding for global variable ‘sim’
  multiplot_proj: no visible binding for global variable ‘variable’
  multiplot_proj: no visible binding for global variable ‘year’
  multiplot_proj: no visible binding for global variable ‘value’
  multiplot_proj: no visible binding for global variable ‘x’
  multiplot_proj: no visible binding for global variable ‘y’
  multiplot_proj: no visible binding for global variable ‘quantile’
  multiplot_proj: no visible binding for global variable ‘blvl’
  multiplot_proj: no visible binding for global variable ‘lo90’
  multiplot_proj: no visible binding for global variable ‘hi90’
  multiplot_proj: no visible binding for global variable ‘lo50’
  multiplot_proj: no visible binding for global variable ‘hi50’
  multiplot_proj: no visible binding for global variable ‘median’
  plot_bycatch_guesses: no visible binding for global variable ‘quantile’
  plot_bycatch_guesses: no visible binding for global variable
    ‘bycatch.rate’
  plot_bycatch_guesses: no visible binding for global variable ‘q50’
  plot_bycatch_guesses: no visible binding for global variable ‘q2.5’
  plot_bycatch_guesses: no visible binding for global variable ‘q97.5’
  plot_bycatch_guesses: no visible binding for global variable ‘q25’
  plot_bycatch_guesses: no visible binding for global variable ‘q75’
  plot_pinnipeds: no visible binding for global variable ‘Value’
  plot_pinnipeds: no visible binding for global variable ‘Range_lo’
  plot_pinnipeds: no visible binding for global variable ‘Range_hi’
  plot_pinnipeds: no visible binding for global variable ‘Species’
  plot_pinnipeds: no visible binding for global variable ‘Parameter’
  plot_pinnipeds: no visible binding for global variable ‘value’
  plot_pinnipeds: no visible binding for global variable ‘Citation’
  plot_pinnipeds: no visible binding for global variable ‘param.mean’
  plot_proj: no visible binding for global variable ‘quantile’
  plot_proj: no visible binding for global variable ‘blvl’
  plot_proj: no visible global function definition for ‘arrange’
  plot_proj: no visible global function definition for ‘desc’
  plot_proj: no visible binding for global variable ‘name’
  plot_proj: no visible binding for global variable ‘sim’
  plot_proj: no visible binding for global variable ‘year’
  plot_proj: no visible binding for global variable ‘value’
  plot_proj: no visible binding for global variable ‘median’
  plot_proj: no visible binding for global variable ‘lo90’
  plot_proj: no visible binding for global variable ‘hi90’
  plot_proj: no visible binding for global variable ‘lo50’
  plot_proj: no visible binding for global variable ‘hi50’
  pop_vs_yield: no visible binding for global variable ‘lh.params1’
  pop_vs_yield: no visible binding for global variable ‘rel1p’
  pop_vs_yield: no visible binding for global variable ‘yield’
  pop_vs_yield: no visible global function definition for ‘lines’
  pop_vs_yield: no visible global function definition for ‘legend’
  Undefined global functions or variables:
    A.w AgeMat AgeMat.w Attempt Bycatch BycatchLvl Citation Code Depl
    K1plus Parameter Range_hi Range_lo S0 S0.w S1plus S1plus.w Species
    Type Value abund10 abund50 abundrel10 abundrel20 abundrel50 arrange
    axis.no blvl button_label bycatch.rate dat depletion desc
    fishing.rate fmax hi50 hi90 high lambdaMax legend lh lh.params1 lines
    lo50 lo90 low median nages nages.w name param.mean prebuild100
    prebuild50 probability q2.5 q25 q50 q75 q97.5 quantile rel1p sim text
    value variable x y year yield z.w zero
  Consider adding
    importFrom("datasets", "lh")
    importFrom("graphics", "legend", "lines", "text")
    importFrom("stats", "median", "quantile")
  to your NAMESPACE file.
```

**Comment by mcsiple** [comment]
Sigh... OK this isn't ideal but I get the same error when I install and try to run the app on my Windows machine. For reasons I do not yet understand, it *does not* load when I try:
```
mmrefpoints::run_app()
```
(throws the same error: `$ operator is invalid for atomic vectors`
But it *does* run when I do:

```
library(mmrefpoints)
run_app()
```

Sorry for all the back and forth but I'm just trying to at least get to a spot where you can run the app and comment on it-- and I'll work on this issue in the background.

**Comment by mcsiple** [comment]
Woo! Fixed with commit 3e9f8e4e . Just needed to add `mmrefpoints` to imports for `run_app()` fn.
